### ISSUE TITLE
Geolocation seems to be stopped but still sends location to the server

### ISSUE THREAD

**Issue by awesomeguy3** [transistorsoft/cordova-background-geolocation-lt#393]
## Your Environment
* Plugin version: latest
* Platform: Android
* OS version: 7.0.1
* Device manufacturer / model: Samsung Galaxy S7
* Cordova version (`cordova -v`): 7.0
* Cordova platform version (`cordova platform ls`): 6.2.3
* Plugin config

 BackgroundGeolocation.configure({
	// Geolocation config
	foregroundService: true,
	desiredAccuracy: 0,
	distanceFilter: 10,
	//locationUpdateInterval: 10000,
	stationaryRadius: 25,
	// Activity Recognition config
	activityRecognitionInterval: 10000,
	stopTimeout: 5555,
	// Application config
	debug: true,  // <-- Debug sounds & notifications.
	stopOnTerminate: false,
	startOnBoot: true,
	schedule: [
		'1-7 9:00-24:00'  // Mon-Sun: 9:00am to 8:00pm TESTING
	],
	url: 'http://' + app.serverAddress + '/set_location',
	method: "POST",
	autoSync: true,
	maxDaysToPersist: 3,
	headers: {  // <-- Optional HTTP headers
		 //"X-FOO": "bar"
	},
	params: {   // <-- Optional HTTP params
		 //"auth_token": "maybe_your_server_authenticates_via_token_YES?",
		user_id: app.user.id
	}
}, function(state) {
	if (!state.enabled) {
	         BackgroundGeolocation.startSchedule();
	}
});

## Expected Behavior
As the foregroundService has been set to true, when I close the app, the icon in the notification tray stays (which is fine), then I set up notifications to stop the service with this part of the code:

BackgroundGeolocation.stopSchedule(function() {
     BackgroundGeolocation.stop();
});

The icon dissapears from the notification tray, but the app sends location anyway.

**Comment by christocracy** [comment]
Logs please

**Comment by awesomeguy3** [comment]
Here are all the error logs (logcat in android studio) from starting the app to stopping the plugin.

[https://pastebin.com/BMnVNj8D](url)

In javascript console there were no errors. Note that I called previously mentioned code for stopping the plugin directly through the console now, as it produces the same result.

**Comment by christocracy** [comment]
There are no messages from the plugin in your log.  Logs from the plugin look like this:

You can filter messages from the plugin like so:
```
$ adb logcat -s TSLocationManager
```

```
07-06 16:08:50.708  9836  9836 I TSLocationManager: ╔═════════════════════════════════════════════
07-06 16:08:50.708  9836  9836 I TSLocationManager: ║ BackgroundGeolocation Service started
07-06 16:08:50.708  9836  9836 I TSLocationManager: ╠═════════════════════════════════════════════
07-06 16:08:50.718  9836  9836 I TSLocationManager: [c.t.l.BackgroundGeolocationService c]
07-06 16:08:50.718  9836  9836 I TSLocationManager:   ✅  Started in foreground
07-06 16:08:50.848  9836  9836 I TSLocationManager: [c.t.l.a.BackgroundGeolocation requestActivityUpdates]
07-06 16:08:50.848  9836  9836 I TSLocationManager:   🎾  Start activity updates: 10000
07-06 16:08:50.878  9836  9836 I TSLocationManager: [c.t.l.BackgroundGeolocationService a]
07-06 16:08:50.878  9836  9836 I TSLocationManager:   🔵  setPace: null → false
07-06 16:08:50.898  9836  9836 I TSLocationManager: [c.t.l.BackgroundGeolocationService j]
07-06 16:08:50.898  9836  9836 I TSLocationManager:   🔴  Stop heartbeat
07-06 16:08:50.928  9836  9836 I TSLocationManager: [c.t.l.BackgroundGeolocationService a]
07-06 16:08:50.928  9836  9836 I TSLocationManager:   🎾  Start heartbeat (60)
07-06 16:08:50.988  3276 10115 D TSLocationManager: [c.t.l.BackgroundGeolocationService onActivityRecognitionResult] still (100%)
07-06 16:08:51.008  3276 10115 D TSLocationManager: [c.t.l.a.BackgroundGeolocation isMainActivityActive] YES
07-06 16:08:51.118  9836  9947 D TSLocationManager: [c.t.l.BackgroundGeolocationService onActivityRecognitionResult] still (100%)
07-06 16:08:51.158  9836  9947 D TSLocationManager: [c.t.l.a.BackgroundGeolocation isMainActivityActive] YES
07-06 16:08:51.158  9836 10119 D TSLocationManager: [c.t.l.LocationService onLocation]
07-06 16:08:51.158  9836 10119 D TSLocationManager: ╔═════════════════════════════════════════════
07-06 16:08:51.158  9836 10119 D TSLocationManager: ║ LocationService: location:1001
07-06 16:08:51.158  9836 10119 D TSLocationManager: ╠═════════════════════════════════════════════
07-06 16:08:51.158  9836 10119 D TSLocationManager: ╟─ 📍  Location[fused 45.519308,-73.617028 acc=10 et=+8d18h13m44s982ms bear=231.0]
07-06 16:08:51.188  9836 10119 I TSLocationManager: [c.t.l.l.TSLocationManager onSingleLocationResult]
07-06 16:08:51.188  9836 10119 I TSLocationManager:   🔵  Acquired motionchange position
07-06 16:08:51.198  9836 10119 D TSLocationManager: [c.t.l.l.TSLocationManager calculateMedianAccuracy] 10.0
07-06 16:08:51.248  9836 10119 D TSLocationManager: [c.t.locationmanager.geofence.a a]
```

**Comment by awesomeguy3** [comment]
Here it is [https://pastebin.com/N5D5r0tW](url)

**Comment by christocracy** [comment]
Everything looks good here.  I see the `BackgroundGeolocationService` destroyed.  There are no further location updates from the plugin received.

```
07-07 15:00:57.655 20282-20282/com.dmnlogistics.mivis D/TSLocationManager: [c.t.l.BackgroundGeolocationService onDestroy]
                                                                           ╔═════════════════════════════════════════════
                                                                           ║ BackgroundGeolocationService destroyed
                                                                           ╠═════════════════════════════════════════════
07-07 15:00:57.671 20282-20282/com.dmnlogistics.mivis I/TSLocationManager: [c.t.l.a.BackgroundGeolocation removeActivityUpdates]
                                                                             🔴  Stop activity updates
07-07 15:00:57.697 20282-20282/com.dmnlogistics.mivis D/TSLocationManager: [c.t.l.a.BackgroundGeolocation isMainActivityActive] YES
07-07 15:00:57.708 20282-20282/com.dmnlogistics.mivis I/TSLocationManager: [c.t.l.BackgroundGeolocationService j]
                                                                             🔴  Stop heartbeat
```

When the plugin requests and receives a location, the logs look like this:
```
07-06 16:08:51.158  9836 10119 D TSLocationManager: ╔═════════════════════════════════════════════
07-06 16:08:51.158  9836 10119 D TSLocationManager: ║ LocationService: location:1001
07-06 16:08:51.158  9836 10119 D TSLocationManager: ╠═════════════════════════════════════════════
07-06 16:08:51.158  9836 10119 D TSLocationManager: ╟─ 📍  Location[fused 45.519308,-73.617028 acc=10 et=+8d18h13m44s982ms bear=231.0]
```

**Comment by awesomeguy3** [comment]
Sorry, I figured out now how to trigger the location update when the device isn't moving. [https://pastebin.com/nWUMBUsd](url) Now you can see here that there is Http Service triggered after destroying BackgroundGeolocation. It happened when I turned off and on GPS (and it sent to my server new locations, but it shouldn't)

**Comment by christocracy** [comment]
Ok, I can reproduce that.  That's a bug.

**Comment by christocracy** [comment]
Actually, I have two apps running (thus responding to `providerchange`).  It was my other app logging location-requests.

However, I see where the problem is.  The issue is *only* when location-services is disabled entirely, the case where:
```
07-07 10:09:52.715 11140 11140 I TSLocationManager: ╔═════════════════════════════════════════════
07-07 10:09:52.715 11140 11140 I TSLocationManager: ║ Location-provider change: false
07-07 10:09:52.715 11140 11140 I TSLocationManager: ╠═════════════════════════════════════════════
07-07 10:09:52.715 11140 11140 I TSLocationManager: ╟─ GPS: false
07-07 10:09:52.715 11140 11140 I TSLocationManager: ╟─ Network: false
```

In this case, since it's impossible to fetch a location, the plugin uses the "last known location" as the `providerchange` location.  This block of code needed to check that the plugin is `enabled` before persisting that "last known location".

**Comment by christocracy** [comment]
This is now fixed in the Premium version.  I planning a latest-stable release here soon.

**Comment by awesomeguy3** [comment]
Ok, thanks.

**Comment by christocracy** [comment]
Released to this free version now in `2.7.4`
